platform/entanglement/src/services/CopyService.js

```diff
@@ -55,19 +55,49 @@ define(
         };
 
         /**
+         * A function used to check if a domain object should be cloned
+         * or not.
+         * @callback platform/entanglement.CopyService~filter
+         * @param {DomainObject} domainObject the object to be cloned
+         * @returns {boolean} true if the object should be cloned; false
+         *          if it should be linked
+         */
+
+        /**
          * Creates a duplicate of the object tree starting at domainObject to
          * the new parent specified.
-         * @param domainObject
-         * @param parent
-         * @param progress
+         *
+         * Any domain objects which cannot be created will not be cloned;
+         * instead, these will appear as links. If a filtering function
+         * is provided, any objects which fail that check will also be
+         * linked instead of cloned
+         *
+         * @param {DomainObject} domainObject the object to duplicate
+         * @param {DomainObject} parent the destination for the clone
+         * @param {platform/entanglement.CopyService~filter} [filter]
+         *        an optional function used to filter out objects from
+         *        the cloning process
          * @returns a promise that will be completed with the clone of
          * domainObject when the duplication is successful.
          */
-        CopyService.prototype.perform = function (domainObject, parent) {
-            var $q = this.$q,
-                copyTask = new CopyTask(domainObject, parent, this.policyService, this.$q);
+        CopyService.prototype.perform = function (domainObject, parent, filter) {
+            var policyService = this.policyService;
+
+            function completeFilter(domainObject) {
```

Comment:
I like this method of allowing the copy task allowing a filter to be specified, but this function is named a bit ambiguously.   It's the "completeFilter" what does that mean?